src/Terrain/MapPerspective.java

```diff
@@ -167,7 +167,7 @@ public boolean isLocationFogged(int x, int y)
   }
   public boolean isFogOn()
   {
-    return (null == viewer) || viewer.gameRules.isFogEnabled;
+    return (null == viewer) || viewer.cos[0].gameRules.isFogEnabled;
```

Comment:
Yeah, it definitely feels weird to be consulting a random CO just to see the game rules. Seems like that should live in the `GameInstance`, or maybe in the `MapMaster/Perspective`? Maybe `Army` even, since that's effectively the CO replacement.